backend/lib/endpoints/posts.js

```diff
@@ -98,6 +100,52 @@ async function routes(app) {
       return updatedPost;
     },
   );
+
+  app.put(
```

Comment:
I'm confused a bit... Liking a post is a PUT request while unliking is a DELETE request?
I'd say DELETE is a better contrast to POST method. :thinking:

Reply:
yes, it's an attempt to stay as close to the REST architectural principles as possible.
POST should create a new item on every request, and it is "non-idempotent" which means if you make the same request 3 times it will create 3 new items.
Whereas, PUT and DELETE are "idempotent" which means if you make the same request 3 times, it will not have a different outcome because it can only change something once.
Here however because of using the Mongo query syntax to check if a post has already been liked I decided to return a `400 Bad Request` in case a post is already liked and you try to like it again ... which may be not 100% according to "REST" but it's just hard to know and follow all these principles perfectly :) 

More info and examples here: https://stackoverflow.com/questions/5665893/the-rest-way-to-check-uncheck-like-unlike-favorite-unfavorite-a-resource

Reply:
Actually, maybe I should return a 404 if you try to "delete" a like that doesn't exist, which may be even closer to REST & HTTP principles.